### ISSUE TITLE
iCmd5250 not working with DirectTransport

### ISSUE THREAD

**Issue by ranga543** [IBM/python-itoolkit#48]
@kadler  I tried with `iCmd5250` but it seems to be throwing some kind of error
`test.py`
```
from itoolkit import *
from itoolkit.transport import DirectTransport

itransport = DirectTransport()
itool = iToolKit()
itool.add(iCmd5250('aspgroup_cmd', 'SETASPGRP ASPGRP(TEST)'))
itool.call(itransport)
cmd_out = itool.dict_out('aspgroup_cmd')

if 'error' in cmd_out:
  print ('test')
  print (cmd_out['error'])
  exit()
```
```
{'aspgroup_cmd': {'error': {...}}, u'error1': u" ?xml version='1.0'? xmlservice /sh /xmlservice ", u'error': u'*BADPARSE'}
```

_Originally posted by @ranga543 in https://github.com/IBM/python-itoolkit/issues/47#issuecomment-588599774_

**Comment by ranga543** [comment]
@kadler any idea ?

**Comment by kadler** [comment]
The error indicates that the XML returned from XMLSERVICE could not be parsed. What does `itool.xml_out()` show?, eg.

```python
itool.call(itransport)
print(itool.xml_out())
```

**Comment by ranga543** [comment]
@kadler No, luck I tried that as well it's giving following output
```
<?xml version="1.0" ?><xmlservice>
<error>*BADPARSE</error>
<error><![CDATA[ ?xml version='1.0'? xmlservice /sh /xmlservice ]]></error>
</xmlservice>
```

**Comment by kadler** [comment]
Could be running in to this problem: https://github.com/IBM/python-itoolkit#ilibcall-and-64-bit-support

You can try installing SI70667 or SI70668.

**Comment by ranga543** [comment]
@kadler after installing `SI70667` PTF we are getting following error
```
sh: syntax error at line 1 : `(' unexpected
<?xml version="1.0" ?><xmlservice><sh error="on" var="aspgroup_cmd">
</sh>
</xmlservice>
```

**Comment by kadler** [comment]
Oh, right. The way `iCmd5250` is programmed, it doesn't support passing parameters...

**Comment by ranga543** [comment]
@kadler I tried without parameters and it's returning
```
from itoolkit import *
from itoolkit.transport import DirectTransport
from itoolkit.lib.ilibcall import *

# modify iToolKit not include row node
itransport = DirectTransport()
#itransport = iLibCall()
itool = iToolKit()
itool.add(iCmd5250('aspgroup_cmd', 'SETASPGRP TEST', {'error': 'on'}))
itool.call(itransport)
cmd_out = itool.xml_out()
print (cmd_out)
```
```
<?xml version="1.0" ?><xmlservice><sh error="on" var="aspgroup_cmd">
</sh>
</xmlservice>
```

**Comment by kadler** [comment]
Calling SETASPGRP will never work as expected with iCmd5250 as this is merely a wrapper around iSh, calling `/QOpenSys/usr/bin/system` to execute the command. This will cause a new job to start, set the ASP group, then exit.

The issue related to quoting and passing arguments has been fixed and will be in version 1.7 when it gets released.
